app/actions/handlers.py

```diff
@@ -49,60 +48,41 @@ async def action_pull_observations(integration, action_config: PullObservationsC
 
     collars_triggered = 0
 
-    for index, base64_file in enumerate(action_config.files):
-        try:
-            # Decode the base64 content
-            decoded_content = base64.b64decode(base64_file)
-
-            # Validate if the file is an XML file
-            try:
-                root = etree.fromstring(decoded_content)
-            except etree.XMLSyntaxError as e:
-                raise client.VectronicXMLParseException(e, f"File {index + 1} is not a valid XML file.")
-
-            # Extract <collar ID> and <key>
-            collar_element = root.find(".//collar")
-            if collar_element is None:
-                raise client.VectronicXMLParseException(Exception("Missing <collar> element in the XML."), "Missing <collar> element in the XML.")
-
-            collar_id = collar_element.get("ID")
-            if not collar_id:
-                raise client.VectronicXMLParseException(Exception("Missing 'ID' attribute in <collar> element."), "Missing 'ID' attribute in <collar> element.")
-
-            key_element = collar_element.find("key")
-            if key_element is None or not key_element.text:
-                raise client.VectronicXMLParseException(Exception("Missing <key> element or its value in the XML."), "Missing <key> element or its value in the XML.")
+    try:
+        # Decode the base64 content
+        collars = json.loads(action_config.files)
 
-            collar_key = key_element.text
+        if not collars:
+            logger.warning(f"No valid collars found for integration ID {integration.id} and action_config {action_config}")
+            return {"status": "success", "collars_triggered": 0}
 
-            logger.info(f"Triggering 'action_fetch_collar_observations' action for collar {collar_id} to extract observations...")
+        for collar in collars:
+            parsed_collar = client.CollarData.parse_obj(collar["parsedData"])
+            logger.info(f"Triggering 'action_fetch_collar_observations' action for collar {parsed_collar.collarID} to extract observations...")
             now = datetime.now(timezone.utc)
             device_state = await state_manager.get_state(
                 integration_id=integration.id,
                 action_id="pull_observations",
-                source_id=collar_id
+                source_id=parsed_collar.collarID
```

Comment:
Relatd to [prev comment](https://github.com/PADAS/gundi-integration-vectronic/pull/3/files#r2297992143), then we could use snake_case field names here.
```suggestion
                source_id=parsed_collar.collar_id
```